src/caches/beliefpropagationcache.jl

```diff
@@ -0,0 +1,240 @@
+default_initial_message(inds_e) = ITensor[denseblocks(delta(inds_e))]
+default_messages(ptn::PartitionedGraph) = Dictionary()
+function default_message_update(contract_list::Vector{ITensor}; kwargs...)
+  return contract_exact(contract_list; kwargs...)
+end
+default_message_update_kwargs() = (; normalize=true, contraction_sequence_alg="optimal")
+@traitfn default_bp_maxiters(g::::(!IsDirected)) = is_tree(g) ? 1 : nothing
```

Comment:
Let's use `maxiter` instead of `maxiters`, that is more common in other Julia packages.